### ISSUE TITLE
How to define train_mask in my own dataset. 

### ISSUE THREAD

**Issue by andac-demir** [rusty1s/pytorch_geometric#1516]
## ❓ Questions & Help

Hi! First of all thanks a lot for this great work.

I’ve tried to build a GCN to train my own data. My data is a 3D tensor of shape (num_graphs, num_nodes, num_features). Each one of these graphs is assigned a label and they all have the same connectivity. So edge indices matrix is the same for all but nodes have different features and I want to learn classifying graphs with different node features. 

I begin testing with a dummy data;
```python
dataset = torch.rand((1000, 16, 250))
```
I define a dummy edge_indices matrix and y (graph targets / labels).
```python
# Graph connectivity structured as a double circular ring
edge_index = torch.tensor([[0, 1], [1, 0], [1, 2], [2, 1],
                               [2, 3], [3, 2], [3, 4], [4, 3],
                               [4, 5], [5, 4], [5, 6], [6, 5],
                               [6, 7], [7, 6], [7, 8], [8, 7],
                               [8, 9], [9, 8], [9, 10], [10, 9],
                               [10, 11], [11, 10], [11, 12], [12, 11],
                               [12, 13], [13, 12], [13, 14], [14, 13],
                               [14, 15], [15, 14]], dtype=torch.long)
# Each graph is assigned a label from 0-3
labels = torch.randint(0, 4, (1000,))
```

Then I try to generate graphs with:

```python
from torch_geometric.data import Data
train_set = []
labels = torch.arange(7)
for i in range(len(dataset.shape[0])):
    graph = Data(x=dataset[i], edge_index=edge_index, y=labels)
    train_set.append(graph)
```

I think this is not the right way to follow because in the example training scripts, data was an object with attributes train, validation and test masks which were used to train GCN. Forward methods don't take lists, but objects.

```python
def forward(self, data):
    x, edge_index = data.x, data.edge_index
    x = self.conv1(x, edge_index)
    x = F.relu(x)
    x = F.dropout(x, training=self.training)
    x = self.conv2(x, edge_index)

model.train()
for epoch in range(200):
    optimizer.zero_grad()
    out = model(data)
    loss = F.nll_loss(out[data.train_mask], data.y[data.train_mask])
    loss.backward()
    optimizer.step()
```

I would appreciate a lot if you could help me with how to handle datasets with multiple graphs so I can train my graphs like Cora dataset from the example in PyG documentation. Thank you!

**Comment by rusty1s** [comment]
So you want to perform graph classification instead of node classification? We have multiple examples for graph classification, see *e.g.* [here](https://github.com/rusty1s/pytorch_geometric/blob/master/examples/mutag_gin.py).

Basically, you assign each `data.y` an integer value that denotes its labels:
```python
graph = Data(..., y=labels[i])
```

**Comment by andac-demir** [comment]
Thanks a lot for the link, this helps!
